File: IntegrationTests/Parser/TypeParsing/TypeParserTests/TypeParserTestBase.cs
// Copyright 2012 rubicon informationstechnologie gmbh
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

using System;
using InjectionCop.Parser.TypeParsing;
using NUnit.Framework;

namespace InjectionCop.IntegrationTests.Parser.TypeParsing.TypeParserTests
{
  public class TypeParserTestBase
  {
    protected TypeParser _typeParser;
    protected readonly string c_InjectionCopRuleId = "IC0001";

    [SetUp]
    public void SetUp()
    {
      _typeParser = new TypeParser ();
    }
  }
}

File: IntegrationTests.VB/Parser/TypeParsing/TypeParserTests/TypeParserTestBase.vb
Imports InjectionCop.Parser.TypeParsing
Imports InjectionCop.Utilities
Imports NUnit.Framework
Imports System

Namespace Parser.TypeParsing.TypeParserTests
	Public Class TypeParserTestBase
		Protected _typeParser As TypeParser

		Protected c_InjectionCopRuleId As String = "IC0001"

		<SetUp()>
		Public Sub SetUp()
			Me._typeParser = New TypeParser()
			Me._typeParser.InitializeBlacklistManager(IntrospectionUtility.TypeNodeFactory(Of TypeParserTestBase)())
		End Sub
	End Class
End Namespace
